### PR TITLE
Refactor csrf token handling using Double Submit Cookie pattern

### PR DESCRIPTION
**Pull request by inyono**
Refactors our csrf token handling using the [Double Submit Cookie pattern](https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie):
- No longer pass "csrf token" down via `window.csrf` (because cashing)
- Instead, pass down a new cookie "CSRF" that contains a pseudorandom value associated with the session (same expiration, same store)
- Validator and form field no longer depend on Zend's default CSRF behavior and have been moved into their own module
- Still: need to prevent caching of pages that contain a form for anonymous users

### CODE REVIEW

**Comment by inyono** [comment]
Should fix #116

File: packages/private/edtr-io/src/plugins/image.tsx [hunk 1]
```diff
@@ -90,52 +91,57 @@ export const validateFile: UploadValidator<FileError[]> = file => {
...
+        if (!e.target) return
+        const formData = new FormData()
+        formData.append('attachment[file]', file)
+        formData.append('type', 'file')
+        formData.append('csrf', getCsrfToken())
+
+        axios
+          .post('/attachment/upload', formData)
+          .then(({ data }) => {
+            console.log(data)
```
Comment (Knorrke) [inline packages/private/edtr-io/src/plugins/image.tsx]: remove this probably?

**Comment by inyono** [comment]
@inyono: check cashing

**Comment by inyono** [comment]
Okay, I have _no_ idea where we specify that `auth/login` shouldn't be cached... I'd say we ignore the potential issue for now (since the login page isn't cached anyways) and deal with it if it's really a problem (e.g. by inserting the value dynamically via Cloudflare Worker / JS / ...)

**Pull request merged**